server/src/main/java/org/tctalent/server/service/db/aws/S3ResourceHelper.java

```diff
@@ -34,292 +51,285 @@
 import org.springframework.web.multipart.MultipartFile;
 import org.tctalent.server.exception.FileDownloadException;
 
-import java.io.ByteArrayInputStream;
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.stream.Collectors;
-
 public class S3ResourceHelper {
-    private static final Logger log = LoggerFactory.getLogger(S3ResourceHelper.class);
-    public static long FILE_PART_SIZE = 5 * 1024 * 1024; // 5MB
-
-    private AmazonS3 amazonS3;
-
-    @Value("${aws.s3.bucketName}")
-    private String s3Bucket;
-
-    private TransferManager transferManager;
-
-    @Autowired
-    public S3ResourceHelper(@Value("${aws.credentials.accessKey}") String accessKey,
-                            @Value("${aws.credentials.secretKey}") String secretKey,
-                            @Value("${aws.s3.region}") String s3Region) {
-        AWSCredentials credentials = new BasicAWSCredentials(accessKey, secretKey);
-        amazonS3 = AmazonS3ClientBuilder.standard().withCredentials(new AWSStaticCredentialsProvider(credentials))
-                .withRegion(s3Region).build();
-        transferManager = TransferManagerBuilder.standard().withS3Client(amazonS3).build();
-    }
-
-    public String getS3Bucket() {
-        return s3Bucket;
-    }
-
-    public File downloadFile(String key) throws FileDownloadException {
-        return downloadFile(s3Bucket, key);
-    }
 
-    public File downloadFile(String bucket,
-                             String key)
-            throws FileDownloadException {
-        File downloadFile = null;
-        try {
-            downloadFile = getTmpFile();
-            Download download = transferManager.download(bucket, key, downloadFile);
-            // blocking wait
-            download.waitForCompletion();
-
-            log.debug("downloaded key {} to {}", key, downloadFile.getAbsolutePath());
-
-        } catch (Exception e) {
-            log.error("Error downloading file to: " + key, e);
-            throw new FileDownloadException("Error downloading file from: " + key, e);
-        }
-        return downloadFile;
+  private static final Logger log = LoggerFactory.getLogger(S3ResourceHelper.class);
+  public static long FILE_PART_SIZE = 5 * 1024 * 1024; // 5MB
+
+  private AmazonS3 amazonS3;
+
+  @Value("${aws.s3.bucketName}")
+  private String s3Bucket;
+
+  private TransferManager transferManager;
+
+  @Autowired
+  public S3ResourceHelper(@Value("${aws.credentials.accessKey}") String accessKey,
+      @Value("${aws.credentials.secretKey}") String secretKey,
+      @Value("${aws.s3.region}") String s3Region) {
+    AWSCredentials credentials = new BasicAWSCredentials(accessKey, secretKey);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If possible, use temporary security credentials (IAM roles) instead of long-term access keys.
Long-term access keys, such as those associated with IAM users and AWS account root users, remain valid until you manually revoke them. However, temporary security credentials obtained through IAM roles and other features of the AWS Security Token Service expire after a short period of time. Use temporary security credentials to help reduce your risk in case credentials are accidentally exposed.
Learn more about [best practices for managing AWS access keys](https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html).